WI #314: `SKU ID`, `SKU Price ID` and `SKU Price Details` clarification (#676)
FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec: specification/columns/skuid.md

# SKU ID

ORIGINAL:
A SKU ID is a unique identifier that defines a provider-supported construct for organizing properties that are common across one or more [*SKU Prices*](#glossary:sku-price). SKU ID can be referenced on a catalog or [*price list*](#glossary:price-list) published by a provider to look up detailed information about the SKU. The composition of the properties associated with the SKU ID may differ across providers. Some providers may not support the [*SKU*](#glossary:sku) construct and instead associate all such properties directly with the *SKU Price*. SKU ID is commonly used for analyzing cost based on *SKU*-related properties above the pricing constructs.

PROPOSED:

> A SKU ID is a provider-specified unique identifier that represents a specific [*SKU*](#glossary:sku) (e.g., billable good or service offering) that can be used or purchased. Each SKU ID represents specific functionality and technical specifications of a single *SKU* that are common across one or more [*SKU Prices*](#glossary:sku-price). SKU ID is consistent across different billing accounts, pricing models (e.g., flat-rate, tiered, committed, spot pricing), and regions for a single provider to facilitate price comparisons for the same functionality, regardless of where the functionality is provided or how it's paid for.

Author: Nit: 
Consider '**quantifiable**/measurable/trackable' instead of 'billable' since some charges (e.g., committed charges) are not necessarily billable. 
I understand that this column does not define what is billable and what is not, so 'billable' is not necessarily incorrect - I just prefer 'quantifiable' 😄

Author: Consider mentioning that SKUs are **priced, tracked, measured, and billed**, rather than **used or purchased**. 

We don't directly use or purchase SKUs; rather, we interact with resources and services, while SKUs represent specific elements of those resources and services that are priced, tracked, measured, and eventually billed.

Reviewer: ```suggestion
A SKU ID is a provider-specified unique identifier that represents a specific [*SKU*](#glossary:sku) (e.g., measurable good or service offering) that can be used or purchased. Each SKU ID represents specific functionality and technical specifications of a single *SKU* that are common across one or more [*SKU Prices*](#glossary:sku-price). SKU ID is consistent across different billing accounts, pricing models (e.g., flat-rate, tiered, committed, spot pricing), and regions for a single provider to facilitate price comparisons for the same functionality, regardless of where the functionality is provided or how it's paid for.
```

> A SKU ID is a provider-specified unique identifier that represents a specific [*SKU*](#glossary:sku) (e.g., billable good or service offering) that can be used or purchased. Each SKU ID represents specific functionality and technical specifications of a single *SKU* that are common across one or more [*SKU Prices*](#glossary:sku-price). SKU ID is consistent across different billing accounts, pricing models (e.g., flat-rate, tiered, committed, spot pricing), and regions for a single provider to facilitate price comparisons for the same functionality, regardless of where the functionality is provided or how it's paid for.

Reviewer 2: ```suggestion
A SKU ID is a provider-specified unique identifier that represents a specific [*SKU*](#glossary:sku) (e.g., billable good or service offering) that can be used or purchased. SKU IDs can represent the common aspects across one or more [*SKU Prices*](#glossary:sku-price). SKU ID should be consistent across different billing accounts, pricing models (e.g., flat-rate, tiered, committed), regions, etc., for a single Provider, when the good or service used or purchased is the same, to facilitate price comparisons for the same functionality, regardless of where the functionality is provided or how it's paid for.
```

I think forcing providers to use the same SKU for things like spot offerings (which have a different set of features -- which forces a different consumption model) because the machine is the same is a bad idea. Its perfectly reasonable to consider the spot version of an VM to be a different SKU since it comes with significantly different features (and is aimed at a significantly different set of use cases). 

To me, it would be like saying a condo should have the same SKU when its up for sale vs up for daily-renewed rent. They may look like the same thing but they come wtih significantly different features and serve different use cases. Now, if someone offered the same condo for rent but said if you rent 1 day at a time for n days, you get a different price, or commit up front for straight 7 days , you get different price, then that makes sense to be the same SKU because you're not getting something different and therefore can be justified that it should be the same SKU.

Reviewer 2: Ultimately, the scope of a SKU is going to be defined by the provider. We just need to help guide while providing the flexibility.

Reviewer 3: I strongly feel we must not imply what/how interruptible offerings should be modeled (same or different SKU). Ultimately, the providers that have this aren't going to change anyway. 

But the argument here is that it impacts availability, but not functionality, so it needs to be the same, is problematic IMO. It's like saying a multi-az offering of a database or a storage solution should be the same SKU as the single AZ (or multi-region vs single region). FOCUS doesn't/shouldn't make this determination. 

As an active user of these interruptible offerings, it is not the same thing as the non-interruptible because you are consciously making a decision to use one or the other based on your architecture, expectations etc.

Reviewer: Your single vs. multi-AZ example is a great one. According to the definition, those should be the same SkuId.

Totally agree that you're consciously making a decision. Same for committing to it or not. But you're still getting the same functionality, just at a different price point. This is in line with the intent of the column. If a provider decides to not follow it, then they won't be conformant with that requirement. That's their call.

Reviewer 4: I agree with @UdamDewaraja, this requirement is a concern because we are asserting that we have definition on what specific constructs or features require a separate SKU or require they are the same.  I believe that this is overreach.

Reviewer: Addressed in a change

SKU ID can be referenced on a catalog or [*price list*](#glossary:price-list) published by a provider to look up detailed information about the *SKU*. The composition of the properties associated with the SKU ID may differ across providers. SKU ID is commonly used for analyzing and comparing costs for the same SKU in different regions, tiers, and more.

NOTES:
- SKU == functionality that you're getting
- SkuId should be the same no matter how the billing account is configured
- SkuId should support comparing before/after negotiations

> - SkuId should support comparing with a public/retail price list

Author: These also apply to both SkuPriceId and SkuPriceDetails, right?

Author: outdated. resolving.

- Concepts that are part of a SKU: Publisher, parent product, pricing unit, size/shape, SLA (dev/test or prod), tech specs
- Concepts that are part of a SKU price: State (preview or GA), SKU region, CD type, term, tier, SKU meter
- External factors that impact price: Billing account type, negotiated discounts, billing/charge period

The SkuId column adheres to the following requirements:

* SkuId MUST be present in a [*FOCUS dataset*](#glossary:FOCUS-dataset) when the provider publishes a SKU list.
* SkuId MUST be of type String.
* SkuId MUST NOT be null when [ChargeClass](#chargeclass) is not "Correction" and [ChargeCategory](#chargecategory) is "Usage" or "Purchase".
* SkuId MUST be null when ChargeCategory is "Tax".
* SkuId MAY be null for all other combinations of ChargeClass and ChargeCategory.
* ~SkuId MUST equal SkuPriceId when a provider does not support an overarching SKU ID construct.~

>   > NOTE: This seems wrong. SkuId is the primary identifier. SkuPriceId is the secondary identifier to account for different pricing options for that SkuId. I don't think we should be defining SkuId based on SkuPriceId. It should be the other way around.

Reviewer 2: But many people have just a SKU Price ID right?

Reviewer: Said another way, those providers only have a SkuId which has no variations, so the SkuPriceId is the same as the SkuId. My logic here is that the world has a well-defined SKU ID concept, which we should be aligning to. The factor we're adding is that those SKU IDs can be sold in different ways, which change the price.

Either way, this is a nit. Keeping or removing it doesn't substantively change anything. I'll keep it.

# SKU ID

A SKU ID is a provider-specified unique identifier that represents a specific [*SKU*](#glossary:sku). SKUs are quantifiable goods or service offerings in a [*FOCUS dataset*](#glossary:FOCUS-dataset) that represent specific functionality and technical specifications. Examples of SKUs include but are not limited to:

* A product license that is purchased or subscribed to.
* Usage of a deployed resource from direct user interaction (e.g., request count).
* Usage by a deployed resource based on the resource's configuration (e.g., running hours, storage space).

Each SKU ID represents a unique set of features that can be sold at different price points or [*SKU Prices*](#glossary:sku-price). SKU ID is consistent across all pricing variations, which may differ based on multiple factors beyond the common functionality and technical specifications. Examples include but are not limited to:

* Date the charge was incurred.
* Provider agreements and contracts.

> * Pricing models (e.g., flat rate, tiered, committed).

Reviewer: ```suggestion
* Pricing tiers (e.g., free tier or volume-based tiers).
* Pricing models (e.g., standard, committed).
* Commitment discount pricing terms (e.g., 1 year, 3 years).
* Negotiated discounts or other contractual terms or conditions.
```

# SKU ID

A SKU ID is a provider-specified unique identifier that represents a specific [*SKU*](#glossary:sku). SKUs are quantifiable goods or service offerings in a [*FOCUS dataset*](#glossary:FOCUS-dataset) that represent specific functionality and technical specifications. Examples of SKUs include but are not limited to:

* A product license that is purchased or subscribed to.
* Usage of a deployed resource from direct user interaction (e.g., request count).
* Usage by a deployed resource based on the resource's configuration (e.g., running hours, storage space).

> Each SKU ID represents a unique set of features can be sold at different price points or [*SKU Prices*](#glossary:sku-price). SKU ID is consistent across all pricing variations, which may differ based on multiple factors beyond the common functionality and technical specifications. Examples include but are not limited to:

Author: Maybe we could mention something like this?:

* Pricing tiers (e.g., free tier or volume-based tiers)
* Pricing models (e.g., standard/on-demand, committed)
* [*commitment discount*](#glossary:commitment-discount) pricing terms (e.g., 1-year, 2-year)
* Negotiated discounts or other contractual terms or conditions

**Note:** Regarding the **pricing model** 

- Some of the values mentioned in the original (tiered and committed) are not mutually exclusive.
- The values in this suggestion closely resemble those in Pricing Category, but I intentionally avoided using that term since we explicitly removed any mention of spot/interruptible instances, which are referenced in the definition of PricingCategory Dynamic.

* Date the charge was incurred.
* Provider agreements and contracts.
* Pricing models (e.g., flat rate, tiered, committed).

> SKU ID should be consistent across pricing variations of a good or service to facilitate price comparisons for the same functionality, like where the functionality is provided or how it's paid for. SKU ID can be referenced on a catalog or [*price list*](#glossary:price-list) published by a provider to look up detailed information about the *SKU*. The composition of the properties associated with the SKU ID may differ across providers. SKU ID is commonly used for analyzing and comparing costs for the same SKU across different price points.

Reviewer 5: Discussed in Apr 1 TF1.  The term "points" here is meant more to convey "datapoints", "attributes", and so on -- not a "price point" per se.  So how about we use the term "details", which aligns with the column name, along with a few examples:

```suggestion
SKU ID should be consistent across pricing variations of a good or service to facilitate price comparisons for the same functionality, like where the functionality is provided or how it's paid for. SKU ID can be referenced on a catalog or [*price list*](#glossary:price-list) published by a provider to look up detailed information about the *SKU*. The composition of the properties associated with the SKU ID may differ across providers. SKU ID is commonly used for analyzing and comparing costs for the same SKU across different price details (e.g., term, tier, location).
```

> SKU ID should be consistent across pricing variations of a good or service to facilitate price comparisons for the same functionality, like where the functionality is provided or how it's paid for. SKU ID can be referenced on a catalog or [*price list*](#glossary:price-list) published by a provider to look up detailed information about the *SKU*. The composition of the properties associated with the SKU ID may differ across providers. SKU ID is commonly used for analyzing and comparing costs for the same SKU across different price points.

Reviewer 5: Discussed with @ijurica and she agrees with my suggestion but would like @flanakin to review and accept this suggestion, as he was in charge of authoring SKU ID.

# SKU ID

A SKU ID is a provider-specified unique identifier that represents a specific [*SKU*](#glossary:sku). SKUs are quantifiable goods or service offerings in a [*FOCUS dataset*](#glossary:FOCUS-dataset) that represent specific functionality and technical specifications. Examples of SKUs include but are not limited to:

* A product license that is purchased or subscribed to.
* Usage of a deployed resource from direct user interaction (e.g., request count).
* Usage by a deployed resource based on the resource's configuration (e.g., running hours, storage space).

Each SKU ID represents a unique set of features that can be sold at different price points or [*SKU Prices*](#glossary:sku-price). SKU ID is consistent across all pricing variations, which may differ based on multiple factors beyond the common functionality and technical specifications. Examples include but are not limited to:

* Date the charge was incurred.

> * Provider agreements and contracts.

Reviewer: ```suggestion
```

# SKU ID

A SKU ID is a provider-specified unique identifier that represents a specific [*SKU*](#glossary:sku). SKUs are quantifiable goods or service offerings in a [*FOCUS dataset*](#glossary:FOCUS-dataset) that represent specific functionality and technical specifications. Examples of SKUs include but are not limited to:

* A product license that is purchased or subscribed to.
* Usage of a deployed resource from direct user interaction (e.g., request count).
* Usage by a deployed resource based on the resource's configuration (e.g., running hours, storage space).

Each SKU ID represents a unique set of features that can be sold at different price points or [*SKU Prices*](#glossary:sku-price). SKU ID is consistent across all pricing variations, which may differ based on multiple factors beyond the common functionality and technical specifications. Examples include but are not limited to:

* Date the charge was incurred.
* Provider agreements and contracts.
* Pricing tiers (e.g., free tier or volume-based tiers).

> * Pricing models (e.g., standard, committed).

Reviewer: ```suggestion
```

> * Geographic region.
> * Service Level Agreements (SLAs).

Reviewer 3: ```suggestion
```

Curious what others think of this. I think SLAs definitely should not be here. It's related to the availability. It's perfectly ok for some providers to sell offerings that have different SLAs as different SKUs. That makes sense. A practitioner shouldn't be deceived by using the same SKU and telling them it's the 'same thing you're charged for, just at a different pricing tier' - no, it's not the same thing. Things with low SLAs are used for particular use cases and vice versa.

Even for region, I'd say its overreach for us to force providers to keep SKUs the same across regions. There's a misconception that people are only using the SKU to do the grouping. Practitioners are using the SKU to look up the properties of the SKU and those properties are far more reliable way to do grouping for various use cases. 

That boils SKU down to a 'thing that I'm being charged for', and the only reason different rows have the same SKU ID is that it's the exact same thing. Whether I paid committed or on-demand, it doesn't impact the 'thing I used or purchased'

* Availability (e.g., spot VMs).

> SKU ID should be consistent across pricing variations of a good or service to facilitate price comparisons for the same functionality, like where the functionality is provided or how it's paid for. SKU ID can be referenced on a catalog or [*price list*](#glossary:price-list) published by a provider to look up detailed information about the *SKU*. The composition of the properties associated with the SKU ID may differ across providers. SKU ID is commonly used for analyzing and comparing costs for the same SKU in different regions, tiers, and more.

Reviewer 3: ```suggestion
SKU ID should be consistent across pricing variations of a good or service to facilitate price comparisons for the same functionality, like where the functionality is provided or how it's paid for. SKU ID can be referenced on a catalog or [*price list*](#glossary:price-list) published by a provider to look up detailed information about the *SKU*. The composition of the properties associated with the SKU ID may differ across providers. SKU ID is commonly used for analyzing and comparing costs for the same SKU in different regions, pricing tiers, and more.
```

The SkuId column adheres to the following requirements:

* SkuId MUST be present in a [*FOCUS dataset*](#glossary:FOCUS-dataset) when the provider supports unit pricing concepts and publishes price lists, publicly or as part of contracting.
* SkuId MUST be of type String.
* SkuId MUST conform to [String Handling](#stringhandling) requirements.
* SkuId nullability is defined as follows:
  * SkuId MUST be null when [ChargeCategory](#chargecategory) is "Tax".
  * SkuId MUST NOT be null when ChargeCategory is "Usage" or "Purchase" and [ChargeClass](#chargeclass) is not "Correction".
  * SkuId MAY be null in all other cases.
* SkuId for a given *SKU* adheres to the following additional requirements:
  * SkuId MUST remain consistent across [*billing accounts*](#glossary:billing-account) or contracts.
  * SkuId MUST remain consistent across [PricingCategory](#pricingcategory) values.
  * SkuId MUST remain consistent regardless of any other factors that might impact the price but do not affect the functionality of the SKU.

> * SkuId MUST be associated with a given [ResourceId](#resourceid) or [ServiceName](#servicename) used or purchased.

Reviewer 3: ```suggestion
* SkuId MUST be associated with a given [*resource*](#glossary:resource) or [*service*](#glossary:service) used or purchased.
```
Much clearer when we refer to the concept resource and service than the actual identifiers which are the values in the columns. ResourceIds may be able to be updated OR reused. The SKU or SKU Price ID are associated with the underlying resource and service not the identifiers

The SkuId column adheres to the following requirements:

* SkuId MUST be present in a [*FOCUS dataset*](#glossary:FOCUS-dataset) when the provider supports unit pricing concepts and publishes price lists, publicly or as part of contracting.
* SkuId MUST be of type String.
* SkuId MUST conform to [String Handling](#stringhandling) requirements.
* SkuId nullability is defined as follows:
  * SkuId MUST be null when [ChargeCategory](#chargecategory) is "Tax".
  * SkuId MUST NOT be null when ChargeCategory is "Usage" or "Purchase" and [ChargeClass](#chargeclass) is not "Correction".
  * SkuId MAY be null in all other cases.
* SkuId MUST equal the SkuId for all other charges for the same *SKU* regardless of factors that impact the price but not the functionality.

>   * SkuId MUST be consistent across billing accounts or agreements.

Reviewer 3: ```suggestion
  * SkuId MUST be consistent across [*billing accounts*](#glossary:BillingAccount) or agreements.
```

>   * SkuId MUST be consistent across [PricingCategory](#pricingcategory) values..

Reviewer 6: ```suggestion
  * SkuId MUST be consistent across [PricingCategory](#pricingcategory) values.
```

>   * SkuId MUST be consistent across [PricingCategory](#pricingcategory) values..

Author: ```suggestion
  * SkuId MUST be consistent across [PricingCategory](#pricingcategory) values.
```

>   * SkuId MUST be consistent across [PricingCategory](#pricingcategory) values..
>   * SkuId MUST be consistent across regions.
>   * SkuId MUST be consistent across service level agreement (SLA) variations..

Reviewer 3: ```suggestion
  * SkuId MUST be consistent across [PricingCategory](#pricingcategory) values.
  * SkuId MUST be consistent across regions.
  * SkuId MUST be consistent across service level agreement (SLA) variations.
```

Author: resolving. addressed in other conversations

>   * SkuId MUST be consistent across service level agreement (SLA) variations..

Reviewer 6: ```suggestion
  * SkuId MUST be consistent across service level agreement (SLA) variations.
```

>   * SkuId MUST be consistent across service level agreement (SLA) variations..

Author: ```suggestion
  * SkuId MUST be consistent across service level agreement (SLA) variations.
```

> * SkuId MUST equal the SkuId for all other charges for the same *SKU* regardless of factors that impact the price but not the functionality.
>   * SkuId MUST be consistent across billing accounts or agreements.
>   * SkuId MUST be consistent across [PricingCategory](#pricingcategory) values..
>   * SkuId MUST be consistent across regions.
>   * SkuId MUST be consistent across service level agreement (SLA) variations..
>   * SkuId MUST be consistent across availability options.

Reviewer: I debated whether to use a one-liner or break this out into separate requirements that are more independently enforceable. I felt separate is better because it allows conformance to be more precise, but I'm happy to go either way. Here's the alternative:
```suggestion
* SkuId MUST equal the SkuId for all other charges for the same *SKU* regardless of billing account type or agreement, [PricingCategory](#pricingcategory), region, SLA, availability, or other factors that impact the price but not the functionality.
```

>   * SkuId MUST be consistent across availability options.

Reviewer 6: What does this mean? 

```
* SkuId MUST remain consistent for a given SKU regardless of any other factors that might impact the price but do not affect the functionality of the SKU.
```

>   * SkuId MUST be consistent across availability options.

Author: due to updated, I'm resolving this conversation and opening a [new one](https://github.com/FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec/pull/676#discussion_r2004174657) with a similar suggestion...

> * SkuId MUST equal the SkuId for all other charges for the same *SKU* regardless of factors that impact the price but not the functionality.
>   * SkuId MUST be consistent across [*billing accounts*](#glossary:billing-account) or agreements.
>   * SkuId MUST be consistent across [PricingCategory](#pricingcategory) values.
>   * SkuId MUST be consistent across regions.
>   * SkuId MUST be consistent across service level agreement (SLA) variations.
>   * SkuId MUST be consistent across availability options.

Author: * **I like this breakdown in nested bullets**
  * When each requirement is placed in its own bullet, it becomes easier for the provider to clearly indicate which specific aspects they conform to or do not conform to.

* **Simplified version of the introductory bullet:**
  * I've noticed that this pattern is used throughout the spec: "{ColumnId} SHOULD/MUST remain consistent over time for a given {OtherColumnId}". Based on that, I suggest using a simplified version of the introductory bullet.
  * I might have oversimplified it 😄 

* **Questions**
  * Do the `availability options` refer to **spot instances**, i.e. **interruptibility**? If yes, I suggest we explicitly call out interruptibility, considering it's already defined in the glossary. On the other hand, availability is already encompassed in the SLA, right? 
  * If `agreements` and `contracts `are synonyms, I suggest we use the term `contracts`, as we already have ContractedCost, ContractedUnitPrice, etc., and `agreement` is not mentioned elsewhere in the spec.
 
  * Should we explicitly mention some of these in SKU Price ID as well? For instance SLA?

```suggestion
* SkuId for a given *SKU* adheres to the following additional requirements:
  * SkuId MUST remain consistent across [*billing accounts*](#glossary:billing-account) or contracts.
  * SkuId MUST remain consistent across [PricingCategory](#pricingcategory) values.
  * SkuId MUST remain consistent across regions.
  * SkuId MUST remain consistent across service level agreement (SLA) variations.
  * SkuId MUST remain consistent regardless of the [*Interruptibility*](#glossary:interruptible) of a resource.
  * SkuId MUST remain consistent regardless of any other factors that might impact the price but do not affect the functionality of the SKU.
```

Author: Copying the question @cnharris10 asked in the [related conversation](https://github.com/FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec/pull/676#discussion_r1993422623)

> What does this mean?
> 
> ```
> * SkuId MUST remain consistent for a given SKU regardless of any other factors that might impact the price but do not affect the functionality of the SKU.
> ```

Reviewer: I like "availability" over "interruptibility". Could we put both "availability or interruptibility" or possibly just use interruptible as an example: "...availability options (e.g., interruptible resources)"?

> * SkuId MUST equal SkuPriceId when a provider does not offer the *SKU* with multiple pricing variations.

Author: I believe this should be a MAY condition rather than a MUST. 

Also, I suggest either not mentioning `pricing variations` or being more explicit about them. For instance, the unit price amount can vary over time, and this is also considered a price variation, but SkuId and SkuPriceId MAY still be equal regardless of that variation.. 
If we simply state 'SkuId MAY equal SkuPriceId.', it would leave it up to the other requirements to determine when they can be equal."

```suggestion
* SkuId MAY equal [SkuPriceId](#SkuPriceId).
```

The SkuId column adheres to the following requirements:

* SkuId MUST be present in a [*FOCUS dataset*](#glossary:FOCUS-dataset) when the provider supports unit pricing concepts and publishes price lists, publicly or as part of contracting.
* SkuId MUST be of type String.
* SkuId MUST conform to [String Handling](#stringhandling) requirements.
* SkuId nullability is defined as follows:
  * SkuId MUST be null when [ChargeCategory](#chargecategory) is "Tax".
  * SkuId MUST NOT be null when ChargeCategory is "Usage" or "Purchase" and [ChargeClass](#chargeclass) is not "Correction".
  * SkuId MAY be null in all other cases.
* SkuId for a given *SKU* adheres to the following additional requirements:
  * SkuId MUST remain consistent across [*billing accounts*](#glossary:billing-account) or contracts.
  * SkuId MUST remain consistent across [PricingCategory](#pricingcategory) values.
  * SkuId MUST remain consistent across regions.
  * SkuId MUST remain consistent across service level agreement (SLA) variations.

>   * SkuId MUST remain consistent regardless of the [*Interruptibility*](#glossary:interruptible) of a resource.

Reviewer: ```suggestion
  * SkuId MUST remain consistent regardless of availability options (e.g., [*interruptible*](#glossary:interruptible) resources).
```

>   * SkuId MUST remain consistent regardless of availability options (e.g., [*interruptible*](#glossary:interruptible) resources).

Reviewer 7: I agree with @UdamDewaraja: I would expect there to be 2 different SkuId's for single and multi-AZ offerings.  This stateme conflicts with line 18 on `same functionality` wherein single AZ vs. multi-AZ is not the same functionality.

I like the consistent language as it implies immutability within a dataset for a Sku, but, this line conflicts and should perhaps we removed or modified.

>   * SkuId MUST remain consistent across regions.
>   * SkuId MUST remain consistent across service level agreement (SLA) variations.
>   * SkuId MUST remain consistent regardless of availability options (e.g., [*interruptible*](#glossary:interruptible) resources).

Reviewer 8: ```suggestion
```

I agree with the conversation here, it feels to me these are the lines that are a step too far. I believe we should make the other changes to improve the clarity of the SkuId and decide on these types of specification requirements in a future release.

Reviewer: This goes against everything we've discussed in TF2 during the 1.2 release. I don't agree, but to move forward, I'm moving this to supporting content. We won't be able to close the feedback we scoped in and will need to revisit in 1.3.

>   * SkuId MUST remain consistent regardless of availability options (e.g., [*interruptible*](#glossary:interruptible) resources).

Reviewer 4: I agree with @UdamDewaraja.

>   * SkuId MUST remain consistent regardless of availability options (e.g., [*interruptible*](#glossary:interruptible) resources).

Reviewer: Moved to supporting content

  * SkuId MUST remain consistent regardless of any other factors that might impact the price but do not affect the functionality of the SKU.

> * SkuId MAY equal [SkuPriceId](#SkuPriceId).

Author: suggesting based on [this discussion](https://github.com/FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec/pull/676/files#r2013384813) on SkuPriceId - SkuId relationship:

```suggestion
* SkuId MUST be associated with a given [ResourceId](#resourceid) or [ServiceName](#servicename) used or purchased.
* SkuId MAY equal [SkuPriceId](#SkuPriceId).
```
